src/test/java/bio/terra/dao/DatasetDaoTest.java

```diff
@@ -167,6 +146,31 @@ public void datasetTest() throws Exception {
                 fromDB.getAssetSpecifications().size(),
                 equalTo(2));
         fromDB.getAssetSpecifications().forEach(this::assertAssetSpecs);
+
+        datasetDao.delete(datasetId);
+    }
+
+    @Test
+    public void primaryKeyTest() throws IOException {
+        UUID datasetId = createDataset("dataset-primary-key.json");
+        Dataset fromDB = datasetDao.retrieve(datasetId);
+        Table variants = fromDB.getTableByName("variant").orElseThrow(IllegalStateException::new);
```

Comment:
I like this pattern..reminds me we should be using more Optionals in the codebase.

Reply:
Should we make `getPrimaryKey()` return `Optional` on `Table`? It'd behave like:
* `Table` instances that map to snapshots always return `Optional.EMPTY`
* `Table` instances that map to datasets always return a non-empty optional. The conversion from dataset request to table model handles converting a missing `primaryKey` field to an empty list.